src/AppNavigator.tsx

```diff
@@ -201,3 +192,53 @@ const useAutoLogout = () => {
     duration: 120 * 1000,
   })
 }
+
+export const useAuthAction = (storage: Storage) => {
+  const strings = useStrings()
+  const {logout} = useAuth()
+  const {authMethod, isLoading: loadingAuthMethod} = useAuthMethod(storage)
+  const {canEnableOsAuth, isLoading: loadingCanEnableOsAuth, refetch} = useCanEnableAuthOs()
+
+  const isLoading = loadingAuthMethod || loadingCanEnableOsAuth
+  const hasTurnedOffOSAuth = !isLoading && !canEnableOsAuth && authMethod?.isOS
+
+  React.useEffect(() => {
+    const appStateSubscription = AppState.addEventListener('change', async (appState) => {
+      if (authMethod?.isOS && appState === 'active' && !hasTurnedOffOSAuth) refetch()
+    })
+    return () => appStateSubscription?.remove()
+  }, [authMethod, refetch, hasTurnedOffOSAuth])
+
+  React.useEffect(() => {
+    if (hasTurnedOffOSAuth) {
+      Alert.alert(strings.biometricsChangeTitle, strings.biometricsChangeMessage)
+
+      logout()
```

Comment:
hooks that return a value should not have side effects

Reply:
it logout the user, to force them to create a PIN, it was designed this way, it won't switch to PIN until the user sets one.

Reply:
why not navigate to create pin?

Reply:
if we navigate to create pin and the user goes back, he will end up in a state where it will fail when trying to submit a tx, then we will have to do some check it all submit components. -> Also it makes sense logically, if you disabled your authentication, the least you can expect is to be logged out.